### PR TITLE
Php 8 Support

### PR DESCRIPTION
**Pull request by NedNisW**
I've walked through the code and tried to apply all 7.4 and 8.0 features if i was aware of them :) 

One important thing about the new tempfiles folder in the tests. 
The Test COnfigTest::testConfigurationEnableCompilation used Vsf-FileSystem, but in a newer PHP-DI Version the code was changed to use rename for moving temp dump file to the destination. Before it was dumped directly in the target file.

Rename can handle only same system wrappers so it is not possible to rename from stream to file on disk. Therefore it needs to be adjusted in the tess.

If something is not correct with the pull request, please let me know. First time contributing ;)

### CODE REVIEW

File: tests/ConfigTest.php [hunk 1]
```diff
@@ -42,56 +46,60 @@ public function testConfigurationDisableAutowire()
...
         self::assertSame([1, 2, 3], $config['b']);
         self::assertSame('d', $config['c']);
     }
 
-    public function testConfigurationEnableCompilation()
+    public function testConfigurationEnableCompilation(): void
     {
-        vfsStream::setup('project');
-
-        $url = vfsStream::url('project');
```
Comment (elie29) [inline tests/ConfigTest.php:63]: $url    = sys_get_temp_dir(); better approach and you don't need to modify gitignore and delete the file: 
check my correction: https://travis-ci.com/github/elie29/zend-di-config/builds/220737669

**Review by elie29 — COMMENTED** [review]
@NedNisW  Hi, you forgot to update Changelog and Readme  files. It is not required I can do it. I added some comments. Thank you a lot

**Comment by NedNisW** [comment]
@elie29 The spaces and anonymous function on a new line are because laminas-coding-standard has changed a lot. If i revert this, we will fail with the phpcs check. Any thoughts on this? 
We can use another coding standard if you're interested.

File: src/Config.php [hunk 2]
```diff
@@ -124,24 +125,25 @@ private function addAutowires(): void
...
+            $this->definitions[$alias] = get((string) $target);
         }
     }
 
     private function addDelegators(): void
     {
         foreach ($this->get('delegators') as $name => $delegators) {
             foreach ($delegators as $delegator) {
-                $previous = $name . '-' . (++$this->delegatorCounter);
+                $previous                     = $name . '-' . ++$this->delegatorCounter;
```
[the remark this answers is not in the document]
Comment (NedNisW) [reply src/Config.php:136]: Unfortunately this and the other ones are part of updated laminas coding standard. How should we continue with this?

[the remark this answers is not in the document]
Comment (elie29) [reply src/Config.php:136]: @NedNisW OK keep as it is for the moment, thx

**Review by elie29 — APPROVED** [review]
Thx @NedNisW  for your job :)

**Pull request merged**